lib/Sema/CSDiag.cpp

```diff
@@ -6223,11 +6223,47 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
   if (diagnoseCallContextualConversionErrors(callExpr, CS->getContextualType()))
     return true;
 
-  // Type check the function subexpression to resolve a type for it if
-  // possible.
-  auto fnExpr = typeCheckChildIndependently(callExpr->getFn());
-  if (!fnExpr)
+  auto originalFnType = CS->getType(callExpr->getFn());
+
+  // Check if there is a structural problem in the function expression
+  // itself, by type-checking it allowing unresolved variables. If
```

Comment:
I had a hard time parsing this. Perhaps "Check if there is a structural problem in the function expression by performing type checking with the option to allow unresolved type variables."